File: Module.Persist.TPM/Model/DTO/PromoROIReport.cs
using System;
using Core.Data;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
using Persist;
using System.Linq;
using Newtonsoft.Json;

namespace Module.Persist.TPM.Model.DTO
{
    public class PromoROIReport : IEntity<Guid>
    {
        public Guid Id { get; set; }
        public bool? InOut { get; set; }
        public int? ClientTreeId { get; set; }
        public int? ClientTreeKeyId { get; set; }

        public int? BaseClientTreeId { get; set; }
        [StringLength(400)]
        public string BaseClientTreeIds { get; set; }
        public bool? NeedRecountUplift { get; set; }

        public DateTimeOffset? LastApprovedDate { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public int? Number { get; set; }
        [StringLength(255)]
        public string Name { get; set; }
        public string ClientHierarchy { get; set; }
        public string ProductHierarchy { get; set; }
        [StringLength(255)]
        public string MechanicComment { get; set; }
        public double? MarsMechanicDiscount { get; set; }
        public double? PlanInstoreMechanicDiscount { get; set; }
        public DateTimeOffset? LastChangedDate { get; set; }
        public DateTimeOffset? LastChangedDateDemand { get; set; }
        public DateTimeOffset? LastChangedDateFinance { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public DateTimeOffset? DispatchesStart { get; set; }
        public DateTimeOffset? DispatchesEnd { get; set; }
        public int? PromoDuration { get; set; }
        public int? DispatchDuration { get; set; }
        public string InvoiceNumber { get; set; }
        [StringLength(20)]
        public string Mechanic { get; set; }
        [StringLength(20)]
        public string MechanicIA { get; set; }
        [StringLength(15)]
        public string MarsStartDate { get; set; }
        [StringLength(15)]
        public string MarsEndDate { get; set; }
        [StringLength(15)]
        public string MarsDispatchesStart { get; set; }
        [StringLength(15)]
        public string MarsDispatchesEnd { get; set; }
        [StringLength(255)]
        public string OtherEventName { get; set; }
        public string EventName { get; set; }
        public string PromoStatusName { get; set; }
        public string PlanInstoreMechanicName { get; set; }
        public string PlanInstoreMechanicTypeName { get; set; }
        public int? CalendarPriority { get; set; }
        public double? PlanPromoTIShopper { get; set; }
        public double? PlanPromoTIMarketing { get; set; }
        public double? PlanPromoBranding { get; set; }
        public double? PlanPromoCost { get; set; }
        public double? TIBasePercent { get; set; }
        public double? PlanPromoBTL { get; set; }
        public double? PlanPromoCostProduction { get; set; }
        public double? PlanPromoUpliftPercent { get; set; }
        public double? PlanPromoIncrementalLSV { get; set; }
        public double? PlanPromoLSV { get; set; }
        public double? PlanPromoPostPromoEffectLSVW1 { get; set; }
        public double? PlanPromoPostPromoEffectLSVW2 { get; set; }
        public double? PlanPromoPostPromoEffectLSV { get; set; }
        public double? PlanPromoROIPercent { get; set; }
        public double? PlanPromoIncrementalNSV { get; set; }
        public double? PlanPromoNetIncrementalNSV { get; set; }
        public double? PlanPromoIncrementalMAC { get; set; }
        public double? PlanPromoXSites { get; set; }
        public double? PlanPromoCatalogue { get; set; }
        public double? PlanPromoPOSMInClient { get; set; }
        public double? PlanPromoCostProdXSites { get; set; }
        public double? PlanPromoCostProdCatalogue { get; set; }
        public double? PlanPromoCostProdPOSMInClient { get; set; }
        public double? ActualPromoXSites { get; set; }
        public double? ActualPromoCatalogue { get; set; }
        public double? ActualPromoPOSMInClient { get; set; }
        public double? ActualPromoCostProdXSites { get; set; }
        public double? ActualPromoCostProdCatalogue { get; set; }
        public double? ActualPromoCostProdPOSMInClient { get; set; }
        public double? PlanPromoBaselineLSV { get; set; }
        public double? PlanPromoIncrementalBaseTI { get; set; }
        public double? PlanPromoNetIncrementalBaseTI { get; set; }
        public double? COGSPercent { get; set; }
        public double? PlanPromoIncrementalCOGS { get; set; }
        public double? PlanPromoNetIncrementalCOGS { get; set; }
        public double? PlanPromoTotalCost { get; set; }
        public double? PlanPromoNetIncrementalLSV { get; set; }
        public double? PlanPromoNetLSV { get; set; }
        public double? PlanPromoNetIncrementalMAC { get; set; }
        public double? PlanPromoIncrementalEarnings { get; set; }
        public double? PlanPromoNetIncrementalEarnings { get; set; }
        public double? PlanPromoNetROIPercent { get; set; }
        public double? PlanPromoNetUpliftPercent { get; set; }
        public string ActualInStoreMechanicName { get; set; }
        public string ActualInStoreMechanicTypeName { get; set; }
        public double? ActualPromoBaselineLSV { get; set; }
        public double? ActualInStoreDiscount { get; set; }
        public double? ActualInStoreShelfPrice { get; set; }
        public double? PlanInStoreShelfPrice { get; set; }
        public double? PCPrice { get; set; }
        public double? ActualPromoIncrementalBaseTI { get; set; }
        public double? ActualPromoNetIncrementalBaseTI { get; set; }
        public double? ActualPromoIncrementalCOGS { get; set; }
        public double? ActualPromoNetIncrementalCOGS { get; set; }
        public double? ActualPromoTotalCost { get; set; }
        public double? ActualPromoNetIncrementalLSV { get; set; }
        public double? ActualPromoNetLSV { get; set; }
        public double? ActualPromoNetIncrementalMAC { get; set; }
        public double? ActualPromoIncrementalEarnings { get; set; }
        public double? ActualPromoNetIncrementalEarnings { get; set; }
        public double? ActualPromoNetROIPercent { get; set; }
        public double? ActualPromoNetUpliftPercent { get; set; }
        public double? PlanPromoBaselineBaseTI { get; set; }
        public double? PlanPromoBaseTI { get; set; }
        public double? PlanPromoNetBaseTI { get; set; }
        public double? PlanPromoNSV { get; set; }
        public double? PlanPromoNetNSV { get; set; }
        public double? ActualPromoBaselineBaseTI { get; set; }
        public double? ActualPromoNetBaseTI { get; set; }
        public double? ActualPromoNSV { get; set; }
        public double? ActualPromoBaseTI { get; set; }
        public double? ActualPromoNetNSV { get; set; }
        public double? ActualPromoTIShopper { get; set; }
        public double? ActualPromoTIMarketing { get; set; }
        public double? ActualPromoBranding { get; set; }
        public double? ActualPromoBTL { get; set; }
        public double? ActualPromoCostProduction { get; set; }
        public double? ActualPromoCost { get; set; }
        public double? ActualPromoUpliftPercent { get; set; }
        public double? ActualPromoIncrementalLSV { get; set; }
        public double? ActualPromoLSVByCompensation { get; set; }
        public double? ActualPromoLSV { get; set; }
        public double? ActualPromoPostPromoEffectLSVW1 { get; set; }
        public double? ActualPromoPostPromoEffectLSVW2 { get; set; }
        public double? ActualPromoPostPromoEffectLSV { get; set; }
        public double? ActualPromoROIPercent { get; set; }
        public double? ActualPromoIncrementalNSV { get; set; }
        public double? ActualPromoNetIncrementalNSV { get; set; }
        public double? ActualPromoIncrementalMAC { get; set; }
        public bool? IsAutomaticallyApproved { get; set; }
        public bool? IsCMManagerApproved { get; set; }
        public bool? IsDemandPlanningApproved { get; set; }
        public bool? IsDemandFinanceApproved { get; set; }
        [StringLength(255)]
        public string Client1LevelName { get; set; }
        [StringLength(255)]
        public string Client2LevelName { get; set; }
        public string ClientName { get; set; }
        public string BrandName { get; set; }
        public string TechnologyName { get; set; }

        public string ProductSubrangesList { get; set; }
        public string MarsMechanicName { get; set; }
        public string MarsMechanicTypeName { get; set; }
        public string ProductTreeObjectIds { get; set; }
        public bool? Calculating { get; set; }
        public string BlockInformation { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/promoroireport/PromoROIReport.js
﻿Ext.define('App.model.tpm.promoroireport.PromoROIReport', {
    extend: 'Ext.data.Model',
    idProperty: 'Id',
    breezeEntityType: 'Promo',
    fields: [
        { name: 'Id', hidden: true },
        { name: 'Number', type: 'int', hidden: false, isDefault: true, isKey: true },
        { name: 'Client1LevelName', type: 'string', hidden: false, isDefault: true },//NA/RKA (1 level)
        { name: 'Client2LevelName', type: 'string', hidden: false, isDefault: true },//Client Group (2 level)
        { name: 'ClientName', type: 'string', hidden: false, isDefault: true },//Client (3 level)
        { name: 'BrandName', type: 'string', hidden: false, isDefault: false },
        { name: 'TechnologyName', type: 'string', hidden: false, isDefault: true },
        { name: 'SubName', type: 'string', hidden: false, isDefault: true },//Product (2 level)
        { name: 'ProductSubrangesList', type: 'string', hidden: false, isDefault: true },//Product (3 level)
        { name: 'MarsMechanicName', type: 'string', hidden: false, isDefault: false },
        { name: 'MarsMechanicTypeName', type: 'string', hidden: true, isDefault: false },
        { name: 'MarsMechanicDiscount', type: 'float', hidden: false, isDefault: false },
        { name: 'MechanicComment', type: 'string', hidden: false, isDefault: false },
        { name: 'StartDate', useNull: true, type: 'date', hidden: false, isDefault: false, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'MarsStartDate', type: 'string', useNull: true, hidden: false, isDefault: true },
        { name: 'EndDate', useNull: true, type: 'date', hidden: false, isDefault: false, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'MarsEndDate', type: 'string', useNull: true, hidden: false, isDefault: false },
        { name: 'BudgetYear', useNull: true, type: 'int', hidden: false, isDefault: false },
        { name: 'PromoDuration', useNull: true, type: 'int', hidden: false, isDefault: false },
        { name: 'EventName', type: 'string', hidden: false, isDefault: true },
        { name: 'PromoStatusName', type: 'string', hidden: false, isDefault: true },
        { name: 'InOut', useNull: true, type: 'bool', hidden: false, isDefault: true },
        { name: 'IsGrowthAcceleration', type: 'bool', hidden: false, isDefault: true },
        { name: 'PlanInstoreMechanicName', type: 'string', type: 'string', hidden: true, isDefault: false },
        { name: 'PlanInstoreMechanicTypeName', type: 'string', type: 'string', hidden: true, isDefault: false },
        { name: 'PlanInstoreMechanicDiscount', type: 'float', hidden: false, isDefault: false },
        { name: 'PlanInStoreShelfPrice', useNull: true, type: 'float', hidden: false, isDefault: false },
        // [{ name: 'PcPrice' }] 19
        { name: 'PCPrice', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBaselineLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoUpliftPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoTIShopper', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoTIMarketing', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoXSites', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCatalogue', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoPOSMInClient', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBranding', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBTL', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCostProduction', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCostProdXSites', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCostProdCatalogue', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCostProdPOSMInClient', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoCost', useNull: true, type: 'float', hidden: false, isDefault: false },
        // [{ name: 'TI Base' }] 36
        { name: 'TIBasePercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        // [{ name: 'COGS' }] 39
        { name: 'COGSPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'COGSTn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalCOGS', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalCOGS', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoTotalCost', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoPostPromoEffectLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBaselineBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalMAC', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalMAC', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalEarnings', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalEarnings', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoROIPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetROIPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetUpliftPercent', useNull: true, type: 'float', hidden: false, isDefault: false },

        //Add TI
        { name: 'ActualAddTIMarketing', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualAddTIShopper', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanAddTIMarketingApproved', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanAddTIShopperCalculated', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanAddTIShopperApproved', useNull: true, type: 'float', hidden: false, isDefault: false },

        { name: 'ActualInStoreMechanicName', type: 'string', hidden: true, isDefault: false },
        { name: 'ActualInStoreMechanicTypeName', type: 'string', hidden: true, isDefault: false },
        { name: 'ActualInStoreDiscount', type: 'float', hidden: false, isDefault: false },
        { name: 'ActualInStoreShelfPrice', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'InvoiceNumber', type: 'string', hidden: false, isDefault: false },
        { name: 'ActualPromoBaselineLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoLSVByCompensation', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoLSV', useNull: true, defaultValue: 0, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoUpliftPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetUpliftPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoTIShopper', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoTIMarketing', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoXSites', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCatalogue', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoPOSMInClient', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoBranding', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoBTL', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCostProduction', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCostProdXSites', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCostProdCatalogue', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCostProdPOSMInClient', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoCost', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalCOGS', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalCOGS', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoTotalCost', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoPostPromoEffectLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoBaselineBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetBaseTI', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetNSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalMAC', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalMAC', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalEarnings', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalEarnings', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoROIPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetROIPercent', useNull: true, type: 'float', hidden: false, isDefault: false },
        //New calculation parameters for ROI
        { name: 'PlanPromoIncrementalMACLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalMACLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalMACLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalMACLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalEarningsLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalEarningsLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalEarningsLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalEarningsLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoROIPercentLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetROIPercentLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoROIPercentLSV', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetROIPercentLSV', useNull: true, type: 'float', hidden: false, isDefault: false },

        { name: 'PromoTypesName', useNull: true, type: 'string', hidden: false, isDefault: false },
        { name: 'SumInvoice', useNull: true, type: 'float', hidden: false, isDefault: false },
        // add new prop
        { name: 'PlanPromoIncrementalCOGSTn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalCOGSTn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalCOGSTn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalCOGSTn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoBaselineVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoIncrementalVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNetIncrementalVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoIncrementalVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNetIncrementalVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'TPMmode', type: 'string', hidden: false, isDefault: false },
        { name: 'IsApolloExport', type: 'bool', hidden: false, isDefault: true },
        { name: 'MLmodel', type: 'bool', hidden: false, isDefault: true },
        { name: 'PlanPromoVolume', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'PlanPromoNSVtn', useNull: true, type: 'float', hidden: false, isDefault: false },
        { name: 'ActualPromoNSVtn', useNull: true, type: 'float', hidden: false, isDefault: false },
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'PromoROIReports',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        },
        extraParams: {
            TPMmode: TpmModes.Prod.alias
        }
    },
});